File: src/test/java/de/schramm/royalbash/controller/service/core/creature/CreatureMock.java
package de.schramm.royalbash.controller.service.core.creature;

import de.schramm.royalbash.controller.service.core.Context;
import de.schramm.royalbash.controller.service.core.Creature;
import de.schramm.royalbash.controller.service.core.Game;
import lombok.Builder;
import lombok.Value;

@Value
@Builder(toBuilder = true)
public class CreatureMock implements Creature {

    private final boolean placeableOnSpot = true;

    private final String id;
    private final String name;
    private final int hitpoints;
    private final int attack;
    private final int cost;

    @Override
    public Creature damage(Creature attacker) {
        return this.toBuilder()
                .hitpoints(hitpoints - attacker.getAttack())
                .build();
    }

    @Override
    public boolean isDead() {
        return hitpoints <= 0;
    }

    @Override
    public Game invoke(Context context) {
        return null;
    }
}

File: src/test/java/de/schramm/royalbash/controller/service/core/card/creature/CreatureMock.kt
package de.schramm.royalbash.controller.service.core.card.creature

import de.schramm.royalbash.controller.service.core.Context
import de.schramm.royalbash.controller.service.core.Creature
import de.schramm.royalbash.controller.service.core.Game

data class CreatureMock(
        override val id: String,
        override val name: String = "Creature Mock",
        override val hitpoints: Int = 0,
        override val attack: Int = 0,
        override val cost: Int = 0
) : Creature {

    override fun damage(amountOfDamage: Int): Creature {
        return this.copy(hitpoints = hitpoints - amountOfDamage)
    }

    override fun invoke(context: Context): Game {
        return context.game
    }
}
